Gems/Meshlets/Code/Source/Meshlets/MeshletsUtilities.cpp

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Atom/RHI/Factory.h>
+#include <Atom/RHI/RHIUtils.h>
+
+#include <Atom/RPI.Public/Shader/Shader.h>
+#include <Atom/RPI.Public/Image/StreamingImage.h>
+#include <Atom/RPI.Reflect/Buffer/BufferAssetView.h>
+
+#include <MeshletsUtilities.h>
+
+namespace AZ
+{
+    namespace Meshlets
+    {
+
+        //!=====================================================================================
+        //!
+        //!                                Utility Functions
+        //!
+        //!=====================================================================================
+
+        // [To Do] examine if most of these functions can become global in RPI
+
+        //! Utility function to generate the Srg given the shader and the desired Srg name to be associated to.
+        //! If several shaders are sharing the same Srg (for example perView, perScene), it is enough to
+        //! create the Srg by associating it with a single shader and since the GPU signature and the data
+        //! are referring to the same shared description (preferable set in an [SrgDeclaration].aszli file)
+        //! The association with all shaders will work properly.
+        Data::Instance<RPI::ShaderResourceGroup> UtilityClass::CreateShaderResourceGroup(
+            Data::Instance<RPI::Shader> shader,
+            const char* shaderResourceGroupId,
+            [[maybe_unused]] const char* moduleName)
+        {
+            Data::Instance<RPI::ShaderResourceGroup> srg = RPI::ShaderResourceGroup::Create(shader->GetAsset(), AZ::Name{ shaderResourceGroupId });
+            if (!srg)
+            {
+                AZ_Error(moduleName, false, "Failed to create shader resource group [%s]", shaderResourceGroupId);
+                return nullptr;
+            }
+            return srg;
+        }
+
+        RHI::BufferViewDescriptor UtilityClass::CreateResourceViewWithDifferentFormat(
+            uint32_t offsetInBytes, uint32_t elementCount, uint32_t elementSize,
+            RHI::Format format, RHI::BufferBindFlags overrideBindFlags)
+        {
+            RHI::BufferViewDescriptor viewDescriptor;
+
+            // In the following line I use the element size and not the size based of the
+            // element format since in the more interesting case of structured buffer, the
+            // size will result in an error.
```

Comment:
CreateStructured is only a shell above what I described and in this utility function I allow to create both using the descriptor I provide. 
You are correct that it might provide some minimal protection but I doubt that since if you take a look at what it does, it is exactly do what I described. Using the utility function I provide I can create all buffers using the same method.